### PR TITLE
Add credits elements function

### PR DESCRIPTION
**Pull request by Miepee**
Adds a function that gives some building blocks for someone to make their own credits spoiler function. Originally written because Fusion needed something like that.

Unsure if and what the best way to test this is. The other spoiler functions don't have any tests, and this will later get indirectly tested by Fusion's PDF.

Example output:
grafik

### CODE REVIEW

**Comment by ThanatosGit** [comment]
I'm not a big fan of adding dead code for "maybe future"™

**Comment by Miepee** [comment]
Talked with this with darkszero already. Plan to add tests soon:tm:. Fusion is guaranteed (technically does already) to use this in its PDF.

**Comment by ThanatosGit** [comment]
I don't know...still not a big fan of it because it's dead code now in this PR and could be added when it's needed + it's like the problem with copy & paste code reversed: Adding a generic function which no one uses except one game.

I haven't looked into it but isn't it possible to refactor something like the `generic_credits` thing to use this? Then it would be used by....uh 3?...games and game specific parts could also access this helper function here directly.

File: randovania/exporter/hints/credits_spoiler.py [hunk 1]
```diff
@@ -74,6 +75,41 @@ def sort_pickup(p: PickupEntry):
...
+    all_patches: dict[int, GamePatches],
+    players_config: PlayersConfiguration,
+) -> dict[str, list]:
+    results = {}
+
+    major_pickup_name_order = {
+        pickup.name: index for index, pickup in enumerate(standard_pickup_configuration.pickups_state.keys())
+    }
+
+    def sort_pickup(p: str) -> (float, str):
```
Comment (ThanatosGit) [inline randovania/exporter/hints/credits_spoiler.py:89]: Syntax error

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

**Comment by Miepee** [comment]
IIRC there was something that was holding me back, but can't remember anymore.

**Comment by duncathan** [comment]
I think this needs to be a proper refactor instead of implemented like this

**Pull request closed without merging**